R/derive_var_trtemfl.R

```diff
@@ -212,27 +291,84 @@ derive_var_trtemfl <- function(dataset,
       ))
     }
     if (ignore_time_for_trt_end) {
-      end_cond <- expr(is.na(!!trt_end_date) |
-        date(!!start_date) <= date(!!trt_end_date) + days(end_window))
+      end_cond <- expr(
+        (is.na(!!trt_end_date) |
+          date(!!start_date) <= date(!!trt_end_date) + days(end_window))
+      )
     } else {
-      end_cond <- expr(is.na(!!trt_end_date) | !!start_date <= !!trt_end_date + days(end_window))
+      end_cond <-
+        expr(
+          (is.na(!!trt_end_date) |
+            !!start_date <= !!trt_end_date + days(end_window))
+        )
     }
   }
 
+
+  # Flagging the single records for both single AE collected data and mutiple AE collected data
+  if (is.null(group_var)) {
+    dataset <- dataset %>%
+      mutate(srfl = "Y")
+  } else {
+    srfl <- get_new_tmp_var(dataset)
+    dataset <- dataset %>%
+      group_by(USUBJID, !!group_var) %>%
+      mutate(srfl = ifelse(row_number() == 1, "Y", NA)) %>%
+      ungroup()
+  }
+
   if (is.null(intensity)) {
     worsening_cond <- expr(FALSE)
   } else {
-    worsening_cond <-
-      expr(!!start_date < !!trt_start_date &
-        (!!initial_intensity < !!intensity | is.na(!!initial_intensity) | is.na(!!intensity)))
+    if (is.null(group_var)) {
+      worsening_cond <-
+        expr(
+          !!start_date < !!trt_start_date &
+            (!!initial_intensity < !!intensity | is.na(!!initial_intensity) |
+              is.na(!!intensity))
+        )
+    } else {
+      prev_intensity <- get_new_tmp_var(dataset)
+      worsen_date <- get_new_tmp_var(dataset)
+
+      dataset <- dataset %>%
+        arrange(USUBJID, !!group_var, !!start_date) %>%
+        group_by(USUBJID, !!group_var) %>%
```

Comment:
Should we add a new argument for subject ID instead of using `USUBJID`?

Reply:
I think we need to use an admiraloption here instead of USUBJID explicitly.

Reply:
I was thinking of adding a new argument to this function - good reminders for using admiraloption. How about this: `derive_var_trtemfl(..., subject_keys = get_admiral_option("subject_keys")[2])`
The documentation for the `subject_keys`  argument could be the same as that from [derive_param_tte](https://pharmaverse.github.io/admiral/reference/derive_param_tte.html). What do you think?

Reply:
Yeah, this option looks good.